api/src/org/labkey/api/dataiterator/NameExpressionDataIterator.java

```diff
@@ -93,7 +100,11 @@ public Object get(int i)
                 curName = StringUtils.isEmpty((String)curName) ? null : curName;
 
             if (curName != null)
+            {
+                if (!_allowUserSpecifiedNames)
+                    getErrors().addRowError(new ValidationException("Manual entry of names has been disabled for this folder. Only naming pattern generated names are allowed."));
```

Comment:
```suggestion
                    getErrors().addRowError(new ValidationException("Manual entry of names has been disabled for this folder. Only naming-pattern-generated names are allowed."));
```
Or, if you're anti-hyphen, "Only names generated by naming patterns are allowed."